src/edu/cmu/cs/sasylf/Proof.java

```diff
@@ -157,69 +209,526 @@ public int getWarningCount() {
 		cacheErrorCount();
 		return warnings;
 	}
-	
+
 	/**
 	 * Analyze the SASyLF code in the reader and return the results.
 	 * @param mf may be null
 	 * @param filename should not be null
-	 * @param id may be null if not interested in checking package/module-name errors
+	 * @param id may be null if not interested in checking package/module-name
+	 *     errors
 	 * @param r contents to parse; must not be null
 	 * @returns Results object of results, never null
 	 */
-	public static Proof parseAndCheck(ModuleFinder mf, String filename, ModuleId id, Reader r) {
-		Proof result = new Proof(filename,id);
-		result.parseAndCheck(mf,r);
+	private final static ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Why sticking a bunch of static fields in this class?  Should this be its own class and you make an instance of it?